src/openfermion/hamiltonians/_hubbard.py

```diff
@@ -117,36 +118,42 @@ def fermi_hubbard(x_dimension, y_dimension, tunneling, coulomb,
     Returns:
         hubbard_model: An instance of the FermionOperator class.
     """
+    tunneling = float(tunneling)
+    coulomb = float(coulomb)
+    chemical_potential = float(chemical_potential)
+    magnetic_field = float(magnetic_field)
+
     # Initialize fermion operator class.
     n_sites = x_dimension * y_dimension
-    if spinless:
-        n_spin_orbitals = n_sites
-    else:
-        n_spin_orbitals = 2 * n_sites
-    hubbard_model = FermionOperator((), 0.0)
-    # select particle-hole symmetry
+    n_spin_orbitals = n_sites
+
+    if not spinless:
+        n_spin_orbitals *= 2
+
+    hubbard_model = FermionOperator.zero()
+
+    # Select particle-hole symmetry
     if particle_hole_symmetry:
         coulomb_shift = FermionOperator((), 0.5)
     else:
-        coulomb_shift = FermionOperator((), 0.0)
+        coulomb_shift = FermionOperator.zero()
 
     # Loop through sites and add terms.
     for site in range(n_sites):
-
         # Add chemical potential and magnetic field terms.
         if chemical_potential and spinless:
-            coefficient = -1. * chemical_potential
+            coefficient = -chemical_potential
             hubbard_model += number_operator(
                 n_spin_orbitals, site, coefficient)
 
         if magnetic_field and spinless:
-            sign = (-1.) ** (site)
+            sign = -1 ** site
```

Comment:
`**` has higher precedence than `-` so you need parentheses around the `-1`.

Reply:
Good catch! I'll fix this and see if I can add a quick test for this.